test-utils/src/lib.rs

```diff
@@ -47,23 +49,32 @@ pub async fn init_core_and_create_wf(test_name: &str) -> (Arc<dyn Core>, String)
     (core, starter.get_task_queue().to_string())
 }
 
-/// Create a core instance that will use the provided history to return a workflow task containing
-/// all of it, for testing replay.
-pub async fn init_core_replay(history: &History) -> Arc<dyn Core> {
-    let mut starter = CoreWfStarter::new_tq_name(TEST_Q);
-    let core_fakehist = init_mock_gateway(
-        starter.core_options.clone(),
-        mock_gateway_from_history(history),
-    )
-    .unwrap();
-    core_fakehist
-        .register_worker(starter.worker_config.clone())
-        .await
-        .unwrap();
-    starter.initted_core = Some(Arc::new(core_fakehist));
-    let core = starter.get_core().await;
-    starter.start_wf().await;
-    core
+/// Create a core instance that can be used for replay. See the [ReplayCore] trait for adding
+/// canned history.
+pub fn init_core_replay(opts: TelemetryOptions) -> ReplayCoreImpl {
```

Comment:
This is a pretty core feature, I wonder if it should be in the primary crate. Is the concern having to enable mocks or some other dependency? I expect every lang to expose this feature.

Reply:
Yeah, enabling mocks adds a good amount of bloat. That said it is true probably every SDK is going to turn it on, so your point is valid for sure. The layout just happens to make this easier for now... but we could potentially move this all inside the normal core package. Different PR for that tho I think